src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -144,20 +142,17 @@ public function supportsResult(string $resourceClass, string $operationName = nu
      */
     public function getResult(QueryBuilder $queryBuilder, string $resourceClass = null, string $operationName = null, array $context = [])
     {
-        $doctrineOrmPaginator = new DoctrineOrmPaginator($queryBuilder, $this->useFetchJoinCollection($queryBuilder));
-        $doctrineOrmPaginator->setUseOutputWalkers($this->useOutputWalkers($queryBuilder));
-
+        $doctrineOrmPaginator = $this->getDoctrinePaginator($queryBuilder);
         $resourceMetadata = null === $resourceClass ? null : $this->resourceMetadataFactory->create($resourceClass);
 
         if ($this->isPartialPaginationEnabled($this->requestStack->getCurrentRequest(), $resourceMetadata, $operationName)) {
-            return new class($doctrineOrmPaginator) extends AbstractPaginator {
-            };
+            return $this->getPartialPaginator($doctrineOrmPaginator);
         }
 
-        return new Paginator($doctrineOrmPaginator);
+        return $this->getPaginator($doctrineOrmPaginator);
     }
 
-    private function isPartialPaginationEnabled(Request $request = null, ResourceMetadata $resourceMetadata = null, string $operationName = null): bool
+    public function isPartialPaginationEnabled(Request $request = null, ResourceMetadata $resourceMetadata = null, string $operationName = null): bool
```

Comment:
Because these methods have a lot of api-platform related parameters I think it's a good idea to make them public. Anyway if not, one would have to rewrite the logic in its extension/dataprovider.